nfhook/nfhook.c

```diff
@@ -72,15 +72,19 @@ static struct nf_hook_ops rk_pre_routing = {
 static int __init erk_init(void)
 {
 	pr_info("NFhook: LKM succefully loaded!\n");
-
-	nf_register_hook(&rk_pre_routing);
+	
+	//nf_register_hook(&rk_pre_routing);
+	//for kernel versions 5.x Use this instead of above
+	nf_register_net_hook(&init_net,&rk_pre_routing);
 
 	return 0;
 }
 
 static void __exit erk_exit(void)
 {
-	nf_unregister_hook(&rk_pre_routing);
+	//nf_unregister_hook(&rk_pre_routing);
+	//for kernel versions 5.x Use this instead of above
+		nf_unregister_net_hook(&init_net,&rk_pre_routing);
 	pr_info("NFhook: LKM succefully unloaded!\n");
 }
```

Comment:
Just to make it clear; if your kernel version is not over 5.x, obviously; change-back the commented out lines; and comment out the newly-added lines.